### PR TITLE
envoy filter proxy version match optimization

### PR DESCRIPTION
**Pull request by rshriram**
alternative way of doing https://github.com/istio/istio/pull/21325

### CODE REVIEW

File: manifests/istio-control/istio-discovery/templates/telemetryv2_1.4.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ spec:
       match:
         context: ANY # inbound, outbound, and gateway
         proxy:
-          proxyVersion: '1\.4.*'
+          proxyVersion: '1.4.*'
```
Comment (howardjohn) [inline manifests/istio-control/istio-discovery/templates/telemetryv2_1.4.yaml]: won't this match `1b4`?

**Review by howardjohn — COMMENTED** [review]
I really don't think we should change the semantics of an existing api, alpha or not. Why cause pain for users? We need to get serious about stability

**Comment by kyessenov** [comment]
I don't think this is really moving us in the right direction. We're waiting on xDS client capabilities feature to make it through envoy, to properly provide semantic capabilities. The fact that it is a regex is a temporary, but useful, workaround.

File: manifests/istio-control/istio-discovery/templates/telemetryv2_1.4.yaml (hunk 1 above)
Comment (rshriram) [reply manifests/istio-control/istio-discovery/templates/telemetryv2_1.4.yaml]: no. See the code change. It is not parsed as regex at all.

**Comment by rshriram** [comment]
>I really don't think we should change the semantics of an existing api, alpha or not. 

the whole point of keeping EnvoyFilter as perpetual alpha is so that we will not be over burdened by these API guarantees. The entire thing is internal detail (proxy version format inclusive). So we have the freedom to change as needed. This is no different from changing the route object name format or the filter chain name format.

**Comment by howardjohn** [comment]
We used to have language to say that EnvoyFilter was a perpetual alpha and
it was recently removed -- now it just says that the XDS config we are
applying patches to is subject to change, it does not say the API itself is
alpha.

**Comment by kyessenov** [comment]
@rshriram Alpha does not mean it's OK to make breaking changes in the middle of the release. We have high-level features (telemetry v2) that depend on this API.

Can you just use caret to optimize the regex (e.g. `^1\.4\.`)?

**Comment by howardjohn** [comment]
Let me see what happens with just the caret

**Comment by howardjohn** [comment]
As mentioned in https://github.com/istio/istio/pull/21325, the caret surprisingly does not improve this much

**Comment by rshriram** [comment]
>@rshriram Alpha does not mean it's OK to make breaking changes in the middle of the release. We have high-level features (telemetry v2) that depend on this API.

this change is backwards compatible - has hacks specifically for telemetry v2 from 1.4 series. The only question is if we will honor other users who may/may not be using this field. It is better to change early than late IMO.

**Comment by howardjohn** [comment]
alternative was merged. thanks for the help

**Pull request closed without merging**